### PR TITLE
LBAC-44: Get Entities Location wise count

### PR DESCRIPTION
**Pull request by VANKINEENITAWRUN**
## Issue I worked on

see https://issues.openmrs.org/browse/LBAC-44

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by HerbertYiga** [comment]
some failing checks  here,you can check the CI logs

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	@ResponseBody
+	@ResponseStatus(value = HttpStatus.OK)
+	public Map<String, Object> getPatientLocationWiseCount() {
+
+		List<Patient> patientList = Context.getPatientService().getAllPatients();
+		Map<String, Integer> locationPatientMap = new HashMap<String, Integer>();
+		Map<String, String > locationNamesMap = getLocationUuidNameMap(locationPatientMap);
+		String locationAttributeUuid = Context.getAdministrationService().getGlobalProperty(LocationBasedAccessConstants.LOCATION_ATTRIBUTE_GLOBAL_PROPERTY_NAME);
+		final PersonAttributeType personAttributeType = Context.getPersonService().getPersonAttributeTypeByUuid(locationAttributeUuid);
+		for(Patient patient : patientList){
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java]: need space between `) {`.

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+			PersonAttribute personAttribute = patient.getPerson().getAttribute(personAttributeType);
+			if(personAttribute != null && locationNamesMap.get(personAttribute.getValue()) != null){
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java]: Fix code style issue.

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+	@RequestMapping(value = "/locationwise-users-count",method = RequestMethod.GET)
+	@ResponseBody
+	@ResponseStatus(value = HttpStatus.OK)
+	public Map<String, Object> getUsersLocationWiseCount() {
+
+		Map<String, Integer> locationUsersMap = new HashMap<String, Integer>();
+		Map<String, String > locationNamesMap = getLocationUuidNameMap(locationUsersMap);
+		List<User> usersList = Context.getUserService().getAllUsers();
+		for(User user : usersList){
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java]: Fix code style issue.

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+			List<String> userLocations = LocationUtils.getUserAccessibleLocationUuids(user);
+			for(String userLocation : userLocations){
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java]: Fix code style issue.

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+				if(StringUtils.isNotBlank(userLocation) && locationNamesMap.get(userLocation) != null){
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java]: Fix code style issue.

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+	public Map<String, Object> getEncountersLocationWiseCount() {
+		List<Patient> patientList = Context.getPatientService().getAllPatients();
+		Map<String, Integer> locationEncounterMap = new HashMap<String, Integer>();
+		Map<String, String > locationNamesMap = getLocationUuidNameMap(locationEncounterMap);
+		Map<Integer,List<Encounter>> encounterMap = Context.getEncounterService().getAllEncounters(new Cohort(patientList));
+		Iterator<Map.Entry<Integer, List<Encounter>>> mapIterator = encounterMap.entrySet().iterator();
+		while (mapIterator.hasNext()) {
+			Map.Entry<Integer, List<Encounter>> entry = mapIterator.next();
+			List<Encounter> encounterList = entry.getValue();
+			for(Encounter encounter : encounterList){
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java]: Fix code style issue.

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+				if(encounter.getLocation()!=null){
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java]: Fix code style issue.

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+		}
+		Map<String, Object> obj = new HashMap<String, Object>();
+		obj.put("results",locationEncounterMap);
+		return obj;
+	}
+
+	private Map<String, String > getLocationUuidNameMap(Map<String, Integer> locationEntityMap){
+		List<Location> locationList = Context.getLocationService().getAllLocations();
+		Map<String, String > locationNamesMap = new HashMap<String, String>();
+		for(Location location : locationList){
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LocationWiseEntityCountController.java]: Fix code style issue.

**Pull request merged**